File: CarbonaraWebAPI/Model/DAO/Station.cs
using System.ComponentModel.DataAnnotations;

namespace CarbonaraWebAPI.Model.DAO
{
    public class Station
    {
        public Station()
        {
        }

        public Station( string name, Address address, int capacity)
        {
          
            Name = name;
            Address = address;
            Capacity = capacity;
        }

        [Key]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public Address Address { get; set; }
        [Required]
        public int Capacity { get; set; }
    }
}

File: CarbonaraAngular/src/app/model/station.ts
import { Address } from "./address";

export class Station {
    constructor(
        public id: number,
        public name: string,
        public address: Address,
        public capacity: number
    )
    {}

    public getStationName() 
    {
        return this.address.street + " " + this.address.number;
    }
}